tests/validators/string.test.ts

```diff
@@ -112,12 +112,29 @@ describe('StringValidator', () => {
 				expect(emailPredicate.parse(input)).toBe(input);
 			});
 
-			test.each(['hi@hello', 'foo@bar', 'foo@bar.com/', 'foo@bar.com/index?search=ok'])('GIVEN %p THEN throws a ConstraintError', (input) => {
+			test.each([
+				'hi@hello',
+				'foo@bar',
+				'foo@bar.com/',
+				'foo@bar.com/index?search=ok',
+				'foo@[object Object].com',
+				'',
+				'without-at-sign',
+				'with@multiple-@-signs',
+				"skywalker@did-you-ever-hear-the-tragedy-of-Darth-Plagueis-The-Wise?-I-thought-not.-It's-not-a-story-the-Jedi-would-tell-you.-It's-a-Sith-legend.-Darth-Plagueis-was-a-Dark-Lord-of-the-Sith,-so-powerful-and-so-wise-he-could-use-the-Force-to-influence-the-midichlorians-to-create-life…-He-had-such-a-knowledge-of-the-dark-side-that-he-could-even-keep-the-ones-he-cared-about-from-dying.-The-dark-side-of-the-Force-is-a-pathway-to-many-abilities-some-consider-to-be-unnatural.-He-became-so-powerful…-the-only-thing-he-was-afraid-of-was-losing-his-power,-which-eventually,-of-course,-he-did.-Unfortunately,-he-taught-his-apprentice-everything-he-knew,-then-his-apprentice-killed-him-in-his-sleep.-Ironic.-He-could-save-others-from-death,-but-not-himself.com",
+				'When-you-look-at-the-dark-side,-careful-you-must-be.-For-the-dark-side-looks-back@master-yoda-swamp.com',
+				'short-account-name@domain-name-that-has-more-than-sixty-three-characters-and-is-then.followed-by-another-segment-that-got-split-by-a-full-stop-symbol.com',
+				`foo@bar.${'a'.repeat(64)}`
+			])('GIVEN %p THEN throws a ConstraintError', (input) => {
 				expectError(
 					() => emailPredicate.parse(input),
 					new ExpectedConstraintError('s.string.email', 'Invalid email address', input, 'expected to be an email address')
 				);
 			});
+
+			test.each([undefined, null])('GIVEN %p THEN throws a ValidationError', (input) => {
```

Comment:
This doesn't touch the constraint, it needs to be in the validator's parse tests, at line 11.